client/sites/components/sites-dataviews/dataview-style.scss

```diff
@@ -31,6 +23,15 @@
 
 	table.dataviews-view-table .dataviews-view-table__row td {
 		border-bottom-color: var(--color-border-secondary);
+
+		// We use combined fields on Site title, url, admin url to form the vertical "site-title" field.
+		// We then combine the icon and site-title to form the horizontal "site" field.
+		// Without this rule the combined field receives justify-content: space-between but we want the icon to be aligned with the text.
+		.components-h-stack {
```

Comment:
`// dataview-overide:`

Reply:
This was easy to reproduce, so I thought I'd create a draft PR to demonstrate https://github.com/WordPress/gutenberg/pull/67226 In other scenarios, we may need to create an issue.

I presume it'd take a while for calypso to update dataviews with the fix. In the interim, what if we approach this differently?

1. Only combine fields horizontally that exist as fields (not with other combined fields).

```js
combinedFields: [
							{
								id: 'site',
								label: __( 'Site' ),
								header: '',
								children: [ 'icon', 'site-title' ],
								direction: 'horizontal',
							},
						],
```

2. Make the `site-title` render function to display the two pieces (vertically).

```js
render: ({field}) => { /* render title + url vertically */ }
```

I see that title+URL is displayed in both layouts (table+list) and I don't see any other place where you'd want only one of them. So that'd presumably work. While not ideal, it's a lot better than the override because it won't break with library updates.

Reply:
https://github.com/WordPress/gutenberg/pull/67226 will be part of Gutenberg 19.8 (`wordpress/dataviews` 4.9.0 if I'm not mistaken).

Reply:
> Make the site-title render function to display the two pieces (vertically).

Will need to continue with this approach I think as list view (when site is selected) doesn't support combined fields.

Reply:
Still needed the override but left extra context about when this can be removed.